pkg/server/server.go

```diff
@@ -394,6 +397,53 @@ func (s *Server) Run(ctx context.Context) error {
 		}
 	}
 
+	if s.cfg.InstallNamespaceScheduler {
+		kubeconfig := clientConfig.DeepCopy()
+		adminConfig, err := clientcmd.NewNonInteractiveClientConfig(*kubeconfig, "admin", &clientcmd.ConfigOverrides{}, nil).ClientConfig()
+		if err != nil {
+			return err
+		}
+
+		kcpClient, err := kcpclient.NewClusterForConfig(adminConfig)
+		if err != nil {
+			return err
+		}
+		crossClusterClient := kcpClient.Cluster(clusterAll)
+
+		kubeClient := kubernetes.NewForConfigOrDie(adminConfig)
+		disco := discovery.NewDiscoveryClientForConfigOrDie(adminConfig)
+		dynClient := dynamic.NewForConfigOrDie(adminConfig)
+
+		csif := kcpexternalversions.NewSharedInformerFactory(crossClusterClient, resyncPeriod)
```

Comment:
nit: share these across controllers?

Reply:
I'm going to file an issue and tackle it after we get this and #302 merged, if that's ok?